test/integration/kubernetes_deploy_test.rb

```diff
@@ -686,4 +686,16 @@ def test_ejson_secrets_respects_no_prune_flag
     ejson_cloud.assert_secret_present('catphotoscom', type: 'kubernetes.io/tls', managed: true)
     ejson_cloud.assert_secret_present('ejson-keys', managed: false)
   end
+
+  def test_invalid_context
+    old_config = ENV['KUBECONFIG']
+    begin
+      ENV['KUBECONFIG'] = File.join(__dir__, '../fixtures/invalid_config.yml')
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all(['Failed to reach server for minikube'])
```

Comment:
I'm not getting the errors I'd expect when I try this test. It seems like there's something wrong with the kubeconfig file itself rather than the server. This test needs to prove that the deploy fails specifically when the server is unavailable.

I see this right now:
![image](https://user-images.githubusercontent.com/4789493/31843559-acfa0a98-b5c1-11e7-8d37-09c9198cabfe.png)


I want to see something like this:
![image](https://user-images.githubusercontent.com/4789493/31843547-94781906-b5c1-11e7-8dc8-f5b9d294ec02.png)

And assert on the bad host/port message in addition to the assertion you have

Reply:
So I looked into this and you were right. There is an issue with the config file. On fixing that I realized that the i/o timeout for the kubectl command is 30s which makes the test fairly time consuming to run. The error is:
`Unable to connect to the server: dial tcp 5.6.7.8:443: i/o timeout`

Should this only be retried once? or just once for the test to make sure the suite doesn't grow in time?

Reply:
Ugh, that's a really good point. That timeout is supposedly configurable with a flag, though the flag doc seems to be lying about the default, so you'd need to confirm:
> `--request-timeout string`           The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")

It seems to me we wouldn't want _any_ test waiting 30s, so it'd be nice if we find a solution that lets us tune the timeout for all of them. One option would be to make `KuberentesDeploy::Kubectl#initialize` take a new `default_timeout:` parameter, and then make `kubectl:` an optional argument to `KubernetesDeploy::Runner#initialize` so that we can create our own instance with a custom timeout in the test setup and pass it in. WDYT?

Reply:
I'll work on that

Reply:
I did what you recommended. So it seems like they weren't lying about the no timeout since with the timeout in place, the error changes.

```
[INFO][2017-10-26 08:53:34 -0400]
[INFO][2017-10-26 08:53:34 -0400]	------------------------------------Phase 1: Initializing deploy------------------------------------
[INFO][2017-10-26 08:53:34 -0400]	All required parameters and files are present
[WARN][2017-10-26 08:53:44 -0400]	The following command failed: kubectl version --context\=minikube --request-timeout\=10s
[WARN][2017-10-26 08:53:44 -0400]	Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

[WARN][2017-10-26 08:53:54 -0400]	The following command failed: kubectl version --context\=minikube --request-timeout\=10s
[WARN][2017-10-26 08:53:54 -0400]	Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

[INFO][2017-10-26 08:53:54 -0400]
[INFO][2017-10-26 08:53:54 -0400]	------------------------------------------Result: FAILURE-------------------------------------------
[FATAL][2017-10-26 08:53:54 -0400]	Failed to reach server for minikube
```